kong/router/atc.lua

```diff
@@ -570,6 +570,19 @@ do
   local tb_concat = table.concat
   local replace_dashes_lower = require("kong.tools.string").replace_dashes_lower
 
+  -- traditional_compatible is case insensitive
+  local tuning_header = string.lower
```

Comment:
The traditional compatible code already uses `lower(xxx)`, why is this still needed?

Reply:
It is used for calculating cache hash.

Reply:
Hmm, if we are already matching by lowercase, then does the cache hash matter? Shouldn't it also be lowercase?

Reply:
Removed `lower`.

Reply:
Add back `tuning_header()` and a test case to verify the insensitive cache calculation.

Reply:
remove `tuning_header()` and update the test case.